### PR TITLE
Added test connection support to email & vault

### PR DESCRIPTION
**Pull request by IdoGafi**

### CODE REVIEW

File: plugins/core-mail/core-mail.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		log.Error(err)
+		return false, []byte("test connection failed" + err.Error())
+	}
+
+	// Settings for SMTP server
+	d := gomail.NewDialer(smtpHost, port, fromEmail, password)
+
+	// Now test the credentials
+	if _, err = d.Dial(); err != nil {
+		return false, []byte("test connection failed")
```
Comment (ozhey) [inline plugins/core-mail/core-mail.go:51]: Still no const :-(

File: plugins/core-mail/core-mail.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	if !ok {
+		return false, []byte("mail connection does not contain smtp host port")
+	}
+
+	port, err := strconv.Atoi(smtpPort)
+	if err != nil {
+		err = fmt.Errorf("provided smtp port is invalid, error: %v", err)
+		log.Error(err)
+		return false, []byte("test connection failed" + err.Error())
+	}
```
Comment (ozhey) [inline plugins/core-mail/core-mail.go:44]: There's no space between failed and provided
```suggestion
	if err != nil {
		return false, []byte("test connection failed: provided smtp port is invalid: " + err.Error())
	}
```

File: plugins/core-mail/core-mail.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+	smtpHost, ok := connection.Data["smtpHost"]
+	if !ok {
+		return false, []byte("mail connection does not contain smtp host server")
+	}
+	smtpPort, ok := connection.Data["smtpPort"]
+	if !ok {
+		return false, []byte("mail connection does not contain smtp host port")
+	}
```
Comment (ozhey) [inline plugins/core-mail/core-mail.go:37]: Let's change it, we really just want to see that the string isn't empty
```suggestion
	fromEmail := connection.Data["email"]
	if fromEmail != "" {
		return false, []byte("mail connection does not contain an email address")
	}

	password := connection.Data["password"]
	if password != "" {
		return false, []byte("mail connection does not contain a password")
	}

	smtpHost := connection.Data["smtpHost"]
	if smtpHost != "" {
		return false, []byte("mail connection does not contain smtp host server")
	}
	smtpPort := connection.Data["smtpPort"]
	if smtpPort != "" {
		return false, []byte("mail connection does not contain smtp host port")
	}
```

File: plugins/vault/vault.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+	res, err := connections.SendTestConnectionRequest(requestURL+"/v1/auth/token/lookup-self", http.MethodGet, nil, connection,
+		p.HandleAuth, nil)
+	if err != nil {
+		return false, []byte(err.Error())
+	}
+
+	if res.StatusCode > http.StatusNoContent {
+		return false, []byte(consts.TestConnectionFailed)
+	}
```
Comment (ozhey) [inline plugins/vault/vault.go:43]: Is there a special reason to test for `> noContent`? Is it really ok that it's anything besides 200?

File: plugins/core-mail/core-mail.go (hunk 1 above)
Comment (IdoGafi) [reply plugins/core-mail/core-mail.go:49]: done

**Review by ozhey — APPROVED** [review]

**Pull request merged**